tests/test_csc.py

```diff
@@ -247,28 +256,33 @@ async def testCantPowerOnDuringCooldownPeriod(self):
                 b".0152rFanSpd3d7\r": b"#01520rFanSpd30000BC\r",
                 b".0153rFanSpd4d9\r": b"#01530rFanSpd40000BE\r",
             }
-            harness.csc.kiloarcModel.cooldownPeriod = 15
-            harness.csc.kiloarcModel.warmupPeriod = 15
-            await harness.remote.cmd_enable.set_start(timeout=STD_TIMEOUT)
-            await harness.remote.cmd_startCooling.set_start(timeout=STD_TIMEOUT)
+
+            await self.remote.cmd_enable.set_start(timeout=STD_TIMEOUT)
+            self.csc.kiloarcModel.cooldownPeriod = 15
+            self.csc.kiloarcModel.warmupPeriod = 15
+            await self.remote.cmd_startCooling.set_start(timeout=STD_TIMEOUT)
             await asyncio.sleep(3)  # Sleep while we wait for chiller to start chilling.
-            await harness.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
+            await self.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
             await asyncio.sleep(5)
-            self.assertEqual(harness.csc.kiloarcModel.component.bulbState, 800)
-            await harness.csc.kiloarcModel.warmup_task
-            await harness.remote.cmd_powerLightOff.set_start(timeout=STD_TIMEOUT)
-            self.assertEqual(harness.csc.kiloarcModel.component.bulbState, 0)
+            self.assertEqual(self.csc.kiloarcModel.component.bulbState, 800)
+            await self.csc.kiloarcModel.warmup_task
+            await self.remote.cmd_powerLightOff.set_start(timeout=STD_TIMEOUT)
+            self.assertEqual(self.csc.kiloarcModel.component.bulbState, 0)
             with salobj.assertRaisesAckError():
-                await harness.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
-            await harness.csc.kiloarcModel.cooldown_task
-            await harness.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
+                await self.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
+            await self.csc.kiloarcModel.cooldown_task
+            await self.remote.cmd_powerLightOn.set_start(timeout=STD_TIMEOUT)
```

Comment:
Why do so many of your tests *end* with `asyncio.sleep(10)`?

If it is needed to avoid some nasty warning at shutdown then please find faster way to avoid the warning, such as killing something in the CSC (`self.csc`) at the end of the test. I would put that something in an override of `make_csc` (which is easy; see https://github.com/lsst-ts/ts_MTMount/blob/10864432a61b5bb68f05599dd8dd575d27773876/tests/test_csc.py#L66-L109 for a simple example). Or you could make it a method and call it manually.